network/gossip/libp2p/network.go

```diff
@@ -157,14 +165,22 @@ func (n *Network) Identity() (map[flow.Identifier]flow.Identity, error) {
 
 // Topology returns the identities of a uniform subset of nodes in protocol state using the topology provided earlier
 func (n *Network) Topology() (map[flow.Identifier]flow.Identity, error) {
-	subset, err := n.top.Subset(n.ids, n.fanout())
-	if err != nil {
-		return nil, fmt.Errorf("failed to derive list of peer nodes to connect to: %w", err)
+	myTopics := engine2.GetTopicsByRole(n.role)
```

Comment:
So we should not be passing in the `role` to the network instead we can query the SubscriptionManager for all the channelIDs that the application has so far subscribed for: https://github.com/onflow/flow-go/blob/master/network/gossip/libp2p/subscriptionManager.go#L14
This way the network is agnostic of the node type.

Reply:
Fixed in https://github.com/onflow/flow-go/pull/83/commits/ab6462affb54298f3909dd81a24196a14195ce30.